### ISSUE TITLE
tools/check_dependencies.sh get the wrong version number

### ISSUE THREAD

**Issue by yzmyyff** [kaldi-asr/kaldi#2736]
In ubuntu 16.04, it shows
```
extras/check_dependencies.sh: g++ (g++-4.8) is not supported.
extras/check_dependencies.sh: You need g++ >= 4.8.3, Apple clang >= 5.0 or LLVM clang >= 3.3.
```
But my g++ is 4.8.5
```
g++ --version
g++ (Ubuntu 4.8.5-4ubuntu2) 4.8.5
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by danpovey** [comment]
I think this is an issue on some older versions of ubuntu where `gcc -dumpversion` doesn't print the patch number.  You can just comment that part of extras/check_depencies.sh to get it to work.  It looks like the output of `g++ --version` is hard to parse as it's very different across platforms.
